From 801dc93b790e9a96aa437a3f344214170c0bd540 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.com>
Date: Tue, 13 Apr 2021 01:07:15 +0100
Subject: [PATCH] avdemux: fix build with FFmpeg 4.4

Direct access to avstream->index_entries was removed
in favour of the newly added avformat_index_get_entry()
and friends.

Fixes https://gitlab.freedesktop.org/gstreamer/gst-libav/-/issues/85

Part-of: <https://gitlab.freedesktop.org/gstreamer/gst-libav/-/merge_requests/129>
---
 ext/libav/gstavdemux.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ext/libav/gstavdemux.c b/ext/libav/gstavdemux.c
index 21b46aa..80a0920 100644
--- a/ext/libav/gstavdemux.c
+++ b/ext/libav/gstavdemux.c
@@ -483,7 +483,11 @@ gst_ffmpegdemux_do_seek (GstFFMpegDemux * demux, GstSegment * segment)
     GST_LOG_OBJECT (demux, "keyframeidx: %d", keyframeidx);
 
     if (keyframeidx >= 0) {
+#if LIBAVFORMAT_VERSION_INT >= AV_VERSION_INT(58,78,0)
+      fftarget = avformat_index_get_entry (stream, keyframeidx)->timestamp;
+#else
       fftarget = stream->index_entries[keyframeidx].timestamp;
+#endif
       target = gst_ffmpeg_time_ff_to_gst (fftarget, stream->time_base);
 
       GST_LOG_OBJECT (demux,
-- 
GitLab

